### PR TITLE
[HiTL] Force ECS instances to upload agent logs to S3 upon termination

### PR DESCRIPTION
**Pull request by snyxan**
# Description

Currently in turk version dashboard the agent logs are collected and uploaded to S3 only when turkers click on the 'END' button. We've add strict restrictions on HIT page to enforce turkers to click on that button but still only ~40% logs are collected in S and we are not sure why it happens,

This PR added some handlers to ECS instance terminations so that they can terminate gracefully -- meaning that they will always collect and upload agent logs to S3 right before they are shut down by AWS. Specifically, a SIGTERM signal is what AWS will send to the ECS docker container first when we issue a command like `ecs.stop_task()`, so we trapped that signal and added a customized handler to do the cleanup work.

## Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] Proposes a change (non-breaking change that isn't necessarily a bug)
- [ ] Refactor
- [ ] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [ ] Documentation only change
- [ ] Datasets Release
- [ ] Models Release

## Type of requested review

- [ ] I want a thorough review of the implementation.
- [x] I want a high level review. 
- [ ] I want a deep design review.

## Before and After

If this PR introduces a change or fixes a bug, please add before and after screenshots (if this is causes a UX change) or before and after examples(this could be plain text, videos etc ) to demonstrate the change.

If this PR introduces a new feature, please also add examples or screenshots demonstrating the feature.

# Testing

Please describe the tests that you ran to verify your changes and how we can reproduce.

# Checklist:

- [ ] I have performed manual end-to-end testing of the feature in my environment.
- [ ] I have added Docstrings and comments to the code.
- [ ] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have added relevant collaborators to review the PR before merge.
- [ ] [Polymetis only] I have verified that all scripts in `tests/scripts` runs on hardware.

### CODE REVIEW

File: droidlet/tools/hitl/utils/allocate_instances.py [hunk 1]
```diff
@@ -320,7 +324,7 @@ def free_ecs_instances(instance_ids):
         "--task_name", type=str, help="Task name of the ecs instance to be requested"
     )
     args = parser.parse_args()
-    instance_ips, instance_ids = request_instance(args.instance_num, args.image_tag, args.task_name)
+    instance_ips, instance_ids = request_instance(args.instance_num, args.image_tag, args.task_name, batch_id="0643")
```
Comment (kavyasrinet) [inline droidlet/tools/hitl/utils/allocate_instances.py:327]: why is the batch id hard coded here?

**Review by kavyasrinet — COMMENTED** [review]
The overall change looks good. Left a nit. How are we doing on the data collection after putting this change in place ?

Also, how has this been tested ?

**Comment by snyxan** [comment]
> The overall change looks good. Left a nit. How are we doing on the data collection after putting this change in place ?

We are getting 100% logs uploaded to S3. We'll see if there's data in those logs (if so, then turkers are doing their job right and the issue of not getting 100% logs in previous run comes from something else, e.g. network, etc. If not, then def turkers found out a way to get around our restrictions somehow)

> Also, how has this been tested ?

I tested on this branch -- spin up instances, interacted with some of them and click on the end button, interacted with some but didn't hit the button, leave some instances alone without interacting with them. At the end, all the logs are shown in S3. So the test passed.

File: droidlet/tools/hitl/utils/allocate_instances.py (hunk 1 above)
Comment (snyxan) [reply droidlet/tools/hitl/utils/allocate_instances.py:327]: Everything under __main__() is test code. I tested with this dummy batch_id to show it works.

**Review by ethan-carlson — APPROVED** [review]
I tested this on my 100 HIT run today and it worked for me!

**Review by kavyasrinet — APPROVED** [review]
Thanks so much for confirming the testing process @ethan-carlson and @snyxan

**Pull request merged**